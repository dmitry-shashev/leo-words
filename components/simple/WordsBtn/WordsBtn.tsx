import { FC } from 'react'

interface Props {
  current: number
  wrong: number
  total: number
  hoverMode: boolean
  onClick: () => void
}

export const WordsBtn: FC<Props> = ({
  current,
  wrong,
  total,
  onClick,
  hoverMode,
}) => {
  let bgColor = ''
  if (hoverMode) {
    bgColor = 'bg-blue-100'
  }
  return (
    <button
      onClick={onClick}
      className={`${bgColor} h-10 px-2 text-xl`}
      type="button"
    >
      <span className="text-emerald-500">{current}</span>
      <span className="mx-1">/</span>
      <span className="text-orange-700">{wrong}</span>
      <span className="mx-1">/</span>
      <span className="text-gray-400">{total}</span>
    </button>
  )
}
