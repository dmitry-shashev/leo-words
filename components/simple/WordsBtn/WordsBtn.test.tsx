import { expect, test, vi } from 'vitest'
import { fireEvent, render } from '@testing-library/react'
import { WordsBtn } from '@/components/simple/WordsBtn/WordsBtn'

test('WordsBtn', () => {
  const onClick = vi.fn()

  const { asFragment, getByRole } = render(
    <WordsBtn
      current={15}
      wrong={6}
      total={1231}
      onClick={onClick}
      hoverMode={false}
    />
  )

  expect(asFragment()).toMatchSnapshot()

  expect(onClick).toBeCalledTimes(0)
  fireEvent.click(getByRole('button'))
  expect(onClick).toBeCalledTimes(1)
})
