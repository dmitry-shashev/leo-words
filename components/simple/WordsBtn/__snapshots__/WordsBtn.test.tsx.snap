// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WordsBtn 1`] = `
<DocumentFragment>
  <button
    aria-label="Words button"
    class=" h-10 px-2 text-xl"
    type="button"
  >
    <span
      aria-label="Counter current"
      class="text-emerald-500"
    >
      15
    </span>
    <span
      class="mx-1"
    >
      /
    </span>
    <span
      aria-label="Counter wrong"
      class="text-orange-700"
    >
      6
    </span>
    <span
      class="mx-1"
    >
      /
    </span>
    <span
      aria-label="Counter total"
      class="text-gray-400"
    >
      1231
    </span>
  </button>
</DocumentFragment>
`;
